pkg/system_api.go

```diff
@@ -166,6 +166,17 @@ func (components Components) Walk() Components {
 	return comps
 }
 
+type RenderComponent struct {
```

Comment:
I created a new struct here. Wasn't sure if I should be using the existing `Component`